backend/de.metas.salescandidate.base/src/main/java/de/metas/ordercandidate/api/OLCandOrderFactory.java

```diff
@@ -294,6 +321,55 @@ else if (orderLines.isEmpty())
 		}
 	}
 
+	private void createCompensationGroups()
+	{
+		groupsToOrderLines.keySet()
+				.stream()
+				.map(groupsToOrderLines::get)
+				.forEach(this::createCompensationGroup);
+
+	}
+
+	private void createCompensationGroup(final List<OrderLineId> orderLineIds)
+	{
+		final Optional<I_C_OrderLine> mainOrderLineInGroupOpt = orderLineIds.stream()
+				.filter(primaryOrderLines::contains)
+				.map(OrderLineId::getRepoId)
+				.map(orderLines::get)
+				.findFirst();
+		final I_C_OrderLine mainOrderLineInGroup = Check.assumePresent(mainOrderLineInGroupOpt, "No order line marked as group compensation line");
```

Comment:
can this assumption fail because of inconsistent data provided by the RST-API caller?

Reply:
It shouldn't, because we first validate the data in `OLCandsProcessorExecutor#process()`.